pre/main.ipynb

```diff
@@ -287,20 +448,40 @@
+      "File \u001b[0;32m~/miniforge3/envs/bspumptest/lib/python3.10/selectors.py:562\u001b[0m, in \u001b[0;36mKqueueSelector.select\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    560\u001b[0m ready \u001b[38;5;241m=\u001b[39m []\n\u001b[1;32m    561\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 562\u001b[0m     kev_list \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_selector\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcontrol\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmax_ev\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    563\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mInterruptedError\u001b[39;00m:\n\u001b[1;32m    564\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m ready\n",
+      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
+     ]
+    }
+   ],
    "source": [
     "@async_step\n",
     "async def docker_compose_up_daemon(inject, event):\n",
-    "    (bitswan_dir, docker_compose_yaml) = event\n",
-    "    cmd = [\"docker-compose\", \"-f\", \"/dev/stdin\", \"build\", \"--pull\"]\n",
+    "    (bitswan_dir, docker_compose_yaml, deployment_info) = event\n",
+    "    services_to_recreate = [service for service, info in deployment_info.items() if info[\"recreate\"]]\n",
+    "\n",
+    "    cmd = [\"docker-compose\", \"-f\", \"/dev/stdin\", \"build\", \"--pull\", \" \".join(services_to_recreate)]\n",
```

Comment:
I'm %99 sure this is wrong. You want to add the services to recreate as a list not join them with `<space>`. Joining them with `<space>` tells docker that it should recreate just one service which has spaces in it's name.

Reply:
Oh you're probably right, I thought that it doesn't matter if you have multiple container names in one position but then you could create one string with whole command, I'll change `join` to `extend` that should be correct right?